src/main/java/eu/dissco/core/datacitepublisher/domain/datacite/UriScheme.java

```diff
@@ -0,0 +1,38 @@
+package eu.dissco.core.datacitepublisher.domain.datacite;
+
+public enum UriScheme {
+  ROR("https://ror.org", "ROR"),
+  HANDLE("https://hdl.handle.net", "Handle"),
+  DOI("https://doi.org", "DOI"),
+  QID(null, "Q Number");
+
+  final String uri;
+  final String schemeName;
+
+  private UriScheme(String uri, String schemeName) {
+    this.uri = uri;
+    this.schemeName = schemeName;
+  }
+
+  public static UriScheme determineScheme(String identifier) {
+    if (identifier.contains("ror")) {
+      return UriScheme.ROR;
+    }
+    if (identifier.contains("hdl")) {
+      return UriScheme.HANDLE;
+    }
+    if (identifier.contains("doi")) {
+      return UriScheme.DOI;
+    }
+    return UriScheme.QID;
```

Comment:
This feels a bit tricky, to assume that if it isn't one of the above then it should be QID. Might be better to check for it and throw an exception if it is something else.

Reply:
So qids are defined as: "a identifier of a data item on Wikidata, comprising the letter "Q" followed by one or more digits.", which is a tricky thing to validate. but assuming incoming QIDs will come with the wikidata uri, then we should be able to match on that.